src/shaka_scheme/system/core/lists.cpp

```diff
@@ -0,0 +1,179 @@
+#include "shaka_scheme/system/core/lists.hpp"
+
+namespace shaka {
+namespace core {
+
+NodePtr cons(NodePtr left, NodePtr right) {
+  return create_node(Data(DataPair(left, right)));
+}
+
+NodePtr car(NodePtr node) {
+  if (node->get_type() != Data::Type::DATA_PAIR) {
+    throw shaka::TypeException(10001, "car(): Data does not hold DataPair");

```

Comment:
You're already using namespace `shaka`. Unless i'm misunderstanding something, the `shaka::` is unnecessary.

Reply:
That's true. Just being explicit here that it's a `shaka::TypeException` and not some other `TypeException` class in the future.